openmc/mgxs/library.py

```diff
@@ -717,3 +725,307 @@ def load_from_file(filename='mgxs', directory='mgxs'):
+                                          xs_type=xs_type,
+                                          subdomains=(id,))
+                # multiplicity requires scatter and nu-scatter
+                if ((('scatter matrix' in self.mgxs_types) and
+                     ('nu-scatter matrix' in self.mgxs_types))):
+                    xsdata.set_multiplicity(
+                        self.all_mgxs[id]['nu-scatter matrix'],
+                        self.all_mgxs[id]['scatter matrix'],
+                        xs_type=xs_type, subdomains=(id,))
+                    xsdata.multiplicity = np.nan_to_num(xsdata.multiplicity)
+                    using_multiplicity = True
+                else:
+                    using_multiplicity = False
+
+                if using_multiplicity:
+                    xsdata.set_scatter(self.all_mgxs[id]['nu-scatter matrix'],
+                                       xs_type=xs_type,
+                                       subdomains=(id,))
+                else:
+                    if 'nu-scatter matrix' in self.mgxs_types:
+                        xsdata.set_scatter(
+                            self.all_mgxs[id]['nu-scatter matrix'],
+                            xs_type=xs_type, subdomains=(id,))
+                        # Since we are not using multiplicity, then
+                        # scattering multiplication (nu-scatter) must be
+                        # accounted for approximately by using an adjusted
+                        # absorption cross section.
+                        # We can not do this with a transport x/s so check
+                        # for that.
+                        if 'total' in self.mgxs_types:
+                            xsdata.absorption = \
+                                np.subtract(xsdata.total,
+                                            np.sum(xsdata.scatter[0, :, :],
+                                                   axis=1))
+                        else:
+                            msg = "Absorption cross section must be " + \
+                                  "provided if using a transport cross" + \
+                                  " section and while not providing a " + \
+                                  "scattering matrix"
+                            raise ValueError(msg)
+                    else:
+                        msg = "No nu-scatter matrix data was provided. " + \
+                              "This means neutron balance cannot be " + \
+                              "achieved since (n,xn) multiplication is " + \
+                              "ignored."
+                        warn(msg)
+                        xsdata.set_scatter(self.all_mgxs[id]['scatter matrix'],
+                                           xs_type=xs_type,
+                                           subdomains=(id,))
+
+                xsdatas.append(xsdata)
+            else:
+                mat_names[id] = {}
+                for nuclide in nuclides:
```

Comment:
This would require that your loop over the domains on line 816 use an enumeration:

```python
for i, domain in enumerate(self.domains):
    ...
```